plugin/core/open.py

```diff
@@ -4,15 +4,58 @@
 from .protocol import DocumentUri
 from .protocol import Range
 from .protocol import RangeLsp
+from .protocol import UINT_MAX
 from .typing import Dict, Tuple, Optional
 from .url import parse_uri
 from .views import range_to_region
+from urllib.parse import unquote, urlparse
 import os
+import re
 import sublime
 import subprocess
+import webbrowser
 
 
 opening_files = {}  # type: Dict[str, Tuple[Promise[Optional[sublime.View]], ResolveFunc[Optional[sublime.View]]]]
+FRAGMENT_PATTERN = re.compile(r'^L?(\d+)(?:,(\d+))?(?:-L?(\d+)(?:,(\d+))?)?')
+
+
+def open_file_uri(
+    window: sublime.Window, uri: DocumentUri, flags: int = 0, group: int = -1
+) -> Promise[Optional[sublime.View]]:
+
+    def parse_fragment(fragment: str) -> RangeLsp:
+        match = FRAGMENT_PATTERN.match(fragment)
+        selection = {'start': {'line': 0, 'character': 0}, 'end': {'line': 0, 'character': 0}}  # type: RangeLsp
+        if match:
+            # Line and column numbers in the fragment are assumed to be 1-based and need to be converted to 0-based
+            # numbers for the LSP Position structure.
+            start_line, start_column, end_line, end_column = [max(0, int(g) - 1) if g else None for g in match.groups()]
+            if start_line:
+                selection['start']['line'] = start_line
+                selection['end']['line'] = start_line
+            if start_column:
+                selection['start']['character'] = start_column
+                selection['end']['character'] = start_column
+            if end_line:
+                selection['end']['line'] = end_line
+                selection['end']['character'] = UINT_MAX
+            if end_column is not None:
+                selection['end']['character'] = end_column
+        return selection
```

Comment:
Does it really make sense to default to 0,0 position when fragment doesn't match the pattern? I would think that we should just ignore it then.

Reply:
Or is it on purpose to create more consistency (always scroll to the top) when opening files?

Reply:
I guess this question is for when the file is already open? Actually I have not considered this. Probably it makes sense indeed to ignore the fragment if it doesn't match the pattern, i.e. keep the current scroll position.

Reply:
https://github.com/sublimelsp/LSP/pull/2049/commits/3e9362742615a3dcbfcba6c81f71e4dbb364d1ff